src/index.ts

```diff
@@ -5,45 +5,80 @@
 //Imports
 import {getInput, info, setSecret} from '@actions/core';
 import getClient from './apollo';
-import {login, publish, setRegistryUrl} from './wapm';
+import {login, publish, setRegistryUrl, tokenLogin} from './wapm';
 
 const main = async () =>
 {
   //Get input
-  const username = getInput('username', {
-    required: true
-  });
+  const username = getInput('username');
+  const password = getInput('password');
+  const userToken = getInput('token');
+  if(userToken) {
+    try {
+      loginWithToken(userToken)
```

Comment:
I would place both `loginWithToken` and `loginWithUsername` in the same try-catch block